### PR TITLE
VyOS: OSPFv2 Resource Module

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

##### SUMMARY

Add new resource module vyos_ospfv2.
Targets model - ansible-network/resource_module_models#82
[vyos_ospfv2]

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

vyos_ospfv2
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

File: plugins/module_utils/network/vyos/utils/utils.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def get_lst_same_for_dicts(want, have, lst):
 
 
 def list_diff_have_only(want_list, have_list):
-    """
+    """ -
```
Comment (danielmellado) [inline plugins/module_utils/network/vyos/utils/utils.py:116]: Is this a typo?

Comment (rohitthakur2590) [reply plugins/module_utils/network/vyos/utils/utils.py:116]: @danielmellado  yeah this is a typo, thank you.

File: plugins/modules/vyos_ospfv2.py [hunk 2]
```diff
@@ -0,0 +1,1981 @@
...
+                    description: Delay (msec) from first change received till SPF calculation.
+                    type: int
+                  initial_holdtime:
+                    description: Initial hold time(msec) between consecutive SPF calculations.
+                    type: int
+                  max_holdtime:
+                    description: maximum hold time (sec).
+                    type: int
+  running_config:
+    description:
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:341]: The description for `running_config` is not correct at the moment. It is similar to what we have in https://docs.ansible.com/ansible/latest/modules/ios_config_module.html#parameter-running_config. But the RM's `running_config` is different from the *_config one. What do you think about updating it to - https://github.com/ansible-collections/cisco.nxos/pull/33/files#diff-8f4ac16689eeae1ccce530dea6d10c9bR51?

File: plugins/modules/vyos_ospfv2.py [hunk 3]
```diff
@@ -0,0 +1,1981 @@
...
+# set protocols ospf parameters abr-type 'cisco'
+# set protocols ospf parameters 'opaque-lsa'
+# set protocols ospf parameters 'rfc1583-compatibility'
+# set protocols ospf parameters router-id '192.0.1.1'
+# set protocols ospf passive-interface 'eth1'
+# set protocols ospf passive-interface 'eth2'
+# set protocols ospf redistribute bgp metric '10'
+# set protocols ospf redistribute bgp metric-type '2'
+#
+- name: Delete single attributes of ospfv2 routes.
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:1648]: Single attribute deletes should not be allowed.

File: plugins/modules/vyos_ospfv2.py [hunk 4]
```diff
@@ -0,0 +1,1981 @@
...
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [
+        ("state", "merged", ("config",)),
+        ("state", "replaced", ("config",)),
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:1965]: `config` is required for `rendered` too

File: plugins/modules/vyos_ospfv2.py [hunk 5]
```diff
@@ -0,0 +1,1981 @@
...
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: vyos_ospfv2
+version_added: 2.10
+short_description: This resource module configures and manages attributes of OSPFv2 routes on VyOS network devices.
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:42]: As discussed, for doc purposes, the short description should only be `OSPFv2 resource module`

**Comment by rohitthakur2590** [comment]
recheck

File: plugins/modules/vyos_ospfv2.py (hunk 5 above)
Comment (trishnaguha) [reply plugins/modules/vyos_ospfv2.py:42]: @NilashishC @rohitthakur2590 The short_description should have the short description and then the keyword OSPFV2 resource module at the end.

File: plugins/modules/vyos_ospfv2.py [hunk 6]
```diff
@@ -0,0 +1,1851 @@
...
+# Using merged
+#
+# Before state:
+# -------------
+#
+# vyos@vyos# run show  configuration commands | grep ospf
+#
+#
+- name: Merge the provided configuration with the exisiting running configuration
+      vyos.vyos.vyos_ospfv2:
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:369]: Looks like an indentation issue?

File: plugins/modules/vyos_ospfv2.py [hunk 7]
```diff
@@ -0,0 +1,1851 @@
...
+# set protocols ospf parameters 'opaque-lsa'
+# set protocols ospf parameters 'rfc1583-compatibility'
+# set protocols ospf parameters router-id '192.0.1.1'
+# set protocols ospf passive-interface 'eth1'
+# set protocols ospf passive-interface 'eth2'
+# set protocols ospf redistribute bgp metric '10'
+# set protocols ospf redistribute bgp metric-type '2'
+
+
+# Using deleted
```
Comment (NilashishC) [inline plugins/modules/vyos_ospfv2.py:1608]: This section may have been repeated twice.

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Pull request merged**